cron/cron.go

```diff
@@ -166,7 +166,13 @@ func startFunc(wg *sync.WaitGroup, exitCtx context.Context, logger *logrus.Entry
 
 			jobWg.Add(1)
 
-			runThisJob := func(cronIteration uint64) {
+			// "nextRun" param added to avoid data race with overlapping jobs
+			// this could be written (to avoid addition of 2nd param) as:
+			//    megaNextRun := nextRun
+			// and then, in runThisJob := ... :
+			//    ...
+			//    fn(megaNextRun, jobLogger)
```

Comment:
Thanks for fixing this! That said, the wording is kinda confusing here, and I had to actually run the race detector to understand what this was talking about. How about:

```
			// `nextRun` will be mutated by the next iteration of
			// this loop, so we cannot simply capture it into the
			// closure here. Instead, we make it a parameter so
			// that it gets copied when `runThisJob` is called.
```

Reply:
Marvelous comment, wondeful explanation! Going to add it to next commit